### PR TITLE
Fix BackupBucket reconciliation error by replacing `StrategicMergePatch` with `MergePatch`

### PR DESCRIPTION
**Pull request by seshachalam-yv**
**How to categorize this PR?**

/area backup
/kind bug

**What this PR does / why we need it:**

This PR fixes an error encountered during the reconciliation when deploying `BackupBucket` resources in the garden cluster with `ProviderConfig`.

The error encountered was:

```
unable to find api field in struct RawExtension for the json field
```

- This happens because `StrategicMergePatch` requires schema information to process fields, but `runtime.RawExtension` lacks schema due to its nature of holding arbitrary JSON data. 
- By switching to `MergePatch`, which treats `runtime.RawExtension` fields as opaque, avoid this error and ensure that `BackupBucket` resources are reconciled correctly when `ProviderConfig` is used.

**Which issue(s) this PR fixes:**

**Special notes for your reviewer:**

Found the bug while testing #10866 for PR https://github.com/gardener/gardener-extension-provider-gcp/pull/906
**Release note:**

```bugfix operator
Fixed an error in `BackupBucket` reconciliation by replacing `StrategicMergePatch` with `MergePatch` to properly handle `runtime.RawExtension` fields.
```

### CODE REVIEW

File: pkg/gardenlet/controller/seed/seed/reconciler_reconcile.go [hunk 1]
```diff
@@ -493,15 +494,22 @@ func deployBackupBucketInGarden(ctx context.Context, k8sGardenClient client.Clie
...
 			},
-			ProviderConfig: seed.Spec.Backup.ProviderConfig,
+			ProviderConfig: &runtime.RawExtension{
+				Raw: func() []byte {
+					if seed.Spec.Backup.ProviderConfig != nil {
+						return seed.Spec.Backup.ProviderConfig.Raw
+					}
+					return nil
+				}(),
+			},
```
Comment (rfranzke) [inline pkg/gardenlet/controller/seed/seed/reconciler_reconcile.go]: This looks odd. Why is this change needed? I don't think we do it this way for similar places, do we?

Comment (seshachalam-yv) [reply pkg/gardenlet/controller/seed/seed/reconciler_reconcile.go]: I agree with your point. The change was indeed not necessary. I have reverted the `ProviderConfig` setting back to its original form.

**Review by rfranzke — COMMENTED** [review]
/lgtm

**Comment by shafeeqes** [comment]
Thanks!
/approve
